src/spac/transformations.py

```diff
@@ -313,7 +400,8 @@ def _validate_transformation_inputs(
         adata: anndata,
         layer: Optional[str] = None,
         associated_table: Optional[str] = None,
-        features: Optional[Union[List[str], str]] = None
+        features: Optional[Union[List[str], str]] = None,
+        label: Optional[str] = None,

```

Comment:
let's roll back the changes to this function. you don't actually call this function in knn_clustering or the unittests

Reply:
We call it on line 157 in knn_clustering, `_validate_transformation_inputs` will call `check_annotation` to ensure annotations are in obs